src/parser.cu

```diff
@@ -480,15 +446,58 @@ Grid parseArgs(int argc, char** argv){
         }
     }
 
+    std::string data_dir = par.sval("data_dir");
+    int dimnum = par.ival("dimnum");
+
     // Setting variables
-    if (stat(par.sval("data_dir").c_str(), &st) == -1) {
-        mkdir(par.sval("data_dir").c_str(), 0700);
+    if (stat(data_dir.c_str(), &st) == -1) {
+        mkdir(data_dir.c_str(), 0700);
     }
 
-    if (par.ival("dimnum") == 2){
+    if (dimnum == 2){
         par.store("zDim", 1);
     }
 
+    // Update values which depend on other values, so that they don't need to be entered in order
+
+    if (par.bval("use_param_file")) {
+        std::string param_file = filecheck(data_dir 
+            + par.sval("param_file"));
+        std::cout << "Input parameter file is " <<  param_file << '\n';
+        par.store("param_file", (std::string)param_file);
+    }
+
+    if (par.bval("read_wfc")) {
+        std::string infile = filecheck(data_dir + "wfc_load");
+        std::string infilei = filecheck(data_dir + "wfci_load");
+        par.store("infile", infile);
+        par.store("infilei", infilei);
+    }
+
+    if (dimnum == 3) {
+        if (par.ival("yDim") == 1){
+            par.store("yDim",256);
+        }
+        if (par.ival("zDim") == 1){
+            par.store("zDim",256);
+        }
+    }
+
+    if (dimnum == 2 && par.ival("yDim") == 1){
```

Comment:
This was previously only executed when dimnum was explicitly set to 2, and not when it defaults to 2.
What exactly is this doing, and why?

Reply:
y and zDim are initially set to 1; however, if someone sets the dimensionality to be 2 or 3D, then we need to catch the case where they wanted higher dimensions, but forgot to explicitly set what they wanted for y and zDim. Here, we just set those to be default values of 256.

It's the same as the above clause, but for 2D.

Reply:
Wouldn't it be clearer to have all dimensions default to `256`, then change the unused ones to `1` afterwards?

Reply:
How do we check if they are unused vs the user explicitly setting the value to be `256`?

Reply:
```If (dimnum < 3) {par.store("zDim", 1);}
if (dimnum < 2) {par.store("yDim", 1);}```

Reply:
Oh sure, that would probably work.

Reply:
This is in the `'c'` case, right? In that case, we have to be sure we can use either flag in any order